README.md

```diff
@@ -1,6 +1,181 @@
+[![Total Downloads](https://img.shields.io/packagist/dt/kirschbaum-development/paragon.svg)](https://packagist.org/packages/kirschbaum-development/paragon)
+[![Actions Status](https://github.com/kirschbaum-development/paragon/actions/workflows/tests.yml/badge.svg)](https://github.com/kirschbaum-development/paragon/actions)
+
+## Requirements
+
+| Laravel Version | Paragon Version |
+|:----------------|:----------------|
+| 11.0            | 1.0.x           |
+
+## Installation
+
+```bash
+composer require kirschbaum-development/paragon
+```
+
+## TypeScript Enums
+
+Enums are a fantastic addition to the PHP-verse but are really lame in the TypeScript-verse. However, it can be annoying trying to get those enum values on the
+front-end of your project. Are you supposed to pass them as a method when returning a view or perhaps via an API? This
+generator solves that problem by scraping your app directory for any and all enums and recreates them as TypeScript
+classes so you can import them directly into your Vue, React, or Svelte front-end!
+
+The simplest way to get started is to run the following command:
+
+```bash
+php artisan paragon:generate-enums
+```
+
+That's it. Now, wherever you may have had enums in your project, "paragons" or near perfect duplicates of those have
+been recreated inside of `resources/js/enums`. Here is an example of the output:
+
+```php
+namespace App\Enums;
+
+enum Status: string
+{
+    case Active = 'active';
+    case Inactive = 'inactive';
+}
+```
+
+```ts
+import Enum from '../Enum.ts';
+
+type StatusDefinition = {
+    name: string;
+    value: string;
+};
+
+class Status extends Enum {
+    protected static items = Object.freeze({
+        Active: Object.freeze({
+            name: 'Active',
+            value: 'active',
+        }),
+        Inactive: Object.freeze({
+            name: 'Inactive',
+            value: 'inactive',
+        }),
+    });
+
+    public static get Active(): AlarmStatusDefinition {
+        return this.items['Active'];
+    }
+
+    public static get Inactive(): AlarmStatusDefinition {
+        return this.items['Inactive'];
+    }
+}
+
+export default Status;
+```
+
+At first glance it appears as though a lot more stuff is happening, but the above generated code allows us to interact
+with the enum in a nearly identical way as in PHP. And you may notice the generated TypeScript class extends the `Enum`
+class. This gives us some underlying functionality that is available to every enum.
+
+Using the above code as the example, here are the TypeScript equivalent method and property calls:
+
+```php PHP API
+use App\Enums\Status;
+
+Status::Active;
+Status::Active->value;
+Status::cases();
+Status::from('active'); 
+Status::tryFrom('active'); 
+```
+
+```ts TypeScript API
+import Status from '@/js/enums/Status.ts';
+
+Status.Active;
+Status.Active.value;
+Status.cases();
+Status.from('active');
+Status.tryFrom('active'); 
+```
```

Comment:
I think I'd put this part above the technical explanation on how the TS class looks like

Reply:
Done